src/main/java/com/softserve/service/CategoryService.java

```diff
@@ -0,0 +1,27 @@
+package com.softserve.service;
+
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+import com.softserve.model.Category;
+import com.softserve.repository.ICategoryRepository;
+
+@Service
+public class CategoryService {
+
+	private ICategoryRepository categoryRepository;
+	
+	public CategoryService(ICategoryRepository categoryRepository) {
+		this.categoryRepository = categoryRepository;
+	}
+	//Searching parent categories
+	public List<Category> findAll(){
```

Comment:
why is findAll equals to finding those that do not have a parent category? I think the method name should refer to it, meaning instead of findAll, it should be something like "findByNoParent" or just re-use the name of the method in repo "findByIdParentCategoryIsNull"